roles/kubernetes-apps/csi_driver/cinder/templates/cinder-csi-nodeplugin.yml.j2

```diff
@@ -108,6 +111,10 @@ spec:
         - name: secret-cinderplugin
           secret:
             secretName: cloud-config
+        - name: ca-certs
```

Comment:
The official templates don't mount these certs: https://github.com/kubernetes/cloud-provider-openstack/blob/master/manifests/cinder-csi-plugin/cinder-csi-nodeplugin.yaml
The driver works without it. Do you know where is it used?

Reply:
Yes it works without it, if you specify `cinder_cacert` or the `OS_CACERT` environment variable. However, in the CCM we already mount host's CA certs in the container so it can use those (https://github.com/kubernetes-sigs/kubespray/pull/5491/files#diff-bbcd23cdeeb81cf46114237ad3f6a0d4R51). It's very convenient for when your OpenStack certificate is signed by a CA that's already present on the host machine (which is the case when you use a public CA or a private CA that you trust on the host).

It's not required to make it work but just a convenience feature. I don't think it hurts to add it, at least as an option. Do you agree?

Reply:
I see, thanks for the explanation. Yeah it's convenient.